playback/service/src/main/java/de/danoeh/antennapod/playback/service/internal/LocalPSMP.java

```diff
@@ -700,25 +700,47 @@ protected void endPlayback(final boolean hasEnded, final boolean wasSkipped,
         abandonAudioFocus();
 
         final Playable currentMedia = media;
-        Playable nextMedia = null;
 
         // we should continue to next episode if we were told to continue and we're allowed to (by sleep timer)
         shouldContinue &= callback.shouldContinueToNextEpisode();
 
         if (shouldContinue) {
-            // Load next episode if previous episode was in the queue and if there
-            // is an episode in the queue left.
-            // Start playback immediately if continuous playback is enabled
-            nextMedia = callback.getNextInQueue(currentMedia);
-            if (nextMedia != null) {
-                callback.onPlaybackEnded(nextMedia.getMediaType(), false);
-                // setting media to null signals to playMediaObject() that
-                // we're taking care of post-playback processing
-                media = null;
-                playMediaObject(nextMedia, false, !nextMedia.localFileAvailable(), isPlaying, isPlaying);
-            }
+            // Move database operations to background thread
```

Comment:
Please move this to an independent PR. This needs some more discussion and we don't want to block the other part of the PR